src/main/java/cz/muni/fi/pv286/mmt/representation/ArrayParser.java

```diff
@@ -0,0 +1,314 @@
+public class ArrayParser extends RepresentationParser {
+
+    public ArrayParser(Options options) {
+        super(options);
+    }
+
+    /**
+     * Splits the given string into a list of top-level tokens.
+     *
+     * @param entry The input string.
+     * @return A list of top-level tokens.
+     * @throws InvalidArrayBracketException If the brackets or parentheses
+     *                                      are mismatched or improperly nested.
+     */
+    public static List<String> topLevelSplitTokens(String entry) throws InvalidArrayBracketException {
+        List<String> tokens = new ArrayList<>();
+        List<Character> stack = new ArrayList<>();
+        int start = 0;
+
+        for (int i = 0; i < entry.length(); i++) {
+            char c = entry.charAt(i);
+
+            if (c == '{' || c == '[' || c == '(') {
+                stack.add(c);
+            } else if (c == '}' || c == ']' || c == ')') {
+                if (stack.isEmpty()) {
+                    throw new InvalidArrayBracketException();
+                }
+                char top = stack.remove(stack.size() - 1);
+                if (c == '}' && top != '{') {
+                    throw new InvalidArrayBracketException();
+                }
+                if (c == ']' && top != '[') {
+                    throw new InvalidArrayBracketException();
+                }
+                if (c == ')' && top != '(') {
+                    throw new InvalidArrayBracketException();
+                }
+            } else if (c == ',' && stack.isEmpty()) {
+                tokens.add(entry.substring(start, i).trim());
+                start = i + 1;
+            }
+        }
+
+        if (start < entry.length()) {
+            tokens.add(entry.substring(start).trim());
+        }
+
+        if (!stack.isEmpty()) {
+            throw new InvalidArrayBracketException();
+        }
+
+        return tokens;
+    }
+
+    @Override
+    protected byte[] parseTo(byte[] bytes) throws IOException {
+        ResultTree parsedTree = translateTo(bytes);
+        return parse(parsedTree);
+    }
+
+    private byte[] translateFrom(ResultTree resultTree) throws InvalidArrayValueException {
+        List<Byte> bytesList = new ArrayList<>();
+        for (ResultTree t : resultTree.getChildren()) {
+            if (!t.getChildren().isEmpty()) {
+                throw new InvalidArrayValueException();
+            } else {
+                bytesList.add(t.getValue()[0]);
+            }
+        }
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        bytesList.stream().mapToInt(Byte::intValue).forEach(byteArrayOutputStream::write);
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    private ResultTree translateTo(byte[] bytes) {
+        ResultTree resultTree = new ResultTree(new ArrayList<>());
+
+        for (byte b : bytes) {
+            var rt = new ResultTree(new byte[] {b});
+            resultTree.getChildren().add(rt);
```

Comment:
**issue:** Modifying private attribute in another class

Making an attribute private does not make sense when you allow modifying its elements from the outside.
**suggestion:** In class `ResultTree` create a public addChild method and use this instead of direct modification.